### PR TITLE
abis/linux: Add various missing ip related structs and defines

### PR DESCRIPTION
**Pull request by Dennisbonke**
Split off from #541 

Part of the mlibc LFS projec

### CODE REVIEW

File: abis/linux/in.h [hunk 1]
```diff
@@ -39,11 +39,34 @@ struct sockaddr_in6 {
...
+	struct in6_addr ipi6_addr;
+	uint32_t ipi6_ifindex;
+};
+
+struct group_req {
+	uint32_t gr_interface;
+	struct sockaddr_storage gr_group;
+};
+
+extern const struct in6_addr in6addr_any;
```
Comment (64) [inline abis/linux/in.h:68]: We already define this in `options/posix/include/netinet/in.h`, so I think this should be removed.

File: abis/linux/in.h [hunk 2]
```diff
@@ -101,4 +124,40 @@ struct ipv6_mreq {
...
+#define IPV6_MULTICAST_LOOP 19
+#define IPV6_JOIN_GROUP 20
+#define IPV6_LEAVE_GROUP 21
+#define IPV6_V6ONLY 26
+#define IPV6_RECVPKTINFO 49
+#define IPV6_PKTINFO 50
+#define IPV6_RECVHOPLIMIT 51
+#define IPV6_HOPLIMIT 52
+
+#define __UAPI_DEF_IN_ADDR      0
```
Comment (64) [inline abis/linux/in.h:148]: It might be nice to add a brief comment why these are needed.

**Review by 64 — CHANGES_REQUESTED** [review]

File: abis/linux/in.h [hunk 3]
```diff
@@ -101,4 +122,41 @@ struct ipv6_mreq {
...
+#define __UAPI_DEF_SOCKADDR_IN  0
+#define __UAPI_DEF_IN_CLASS     0
+#define __UAPI_DEF_IN6_ADDR     0
+#define __UAPI_DEF_IN6_ADDR_ALT 0
+#define __UAPI_DEF_SOCKADDR_IN6 0
+#define __UAPI_DEF_IPV6_MREQ    0
+#define __UAPI_DEF_IPPROTO_V6   0
+#define __UAPI_DEF_IPV6_OPTIONS 0
+#define __UAPI_DEF_IN6_PKTINFO  0
+#define __UAPI_DEF_IP6_MTUINFO  0
```
Comment (ArsenArsen) [inline abis/linux/in.h:160]: we should just use the kernel definitions instead (on linux at least)

**Review by ArsenArsen — COMMENTED** [review]
we should consider this practice in a wider sense across the tree

**Comment by avdgrinten** [comment]
Before we add any more `__UAPI` compatibility we should set up a policy to decide (i) what we take from UAPI and what we don't, (ii) what we do on non-Linux systems (seems a bit pointless to rely on the UAPI headers on the one hand, while also duplicating them for all other OSes?) and we should document that properly, see #582.

After this is properly decided and documented, we should go through the existing headers and clean them up.

**Comment by Dennisbonke** [comment]
So what is the status on this one? I'm lost on what the actual consensus is now, can we merge this or is there anything I need to do?

**Comment by Dennisbonke** [comment]
So, can we merge this one then?

**Review by no92 — APPROVED** [review]

**Review by ArsenArsen — APPROVED** [review]
I think this is OK, all things considered, right?

**Review by qookei — APPROVED** [review]
Seems good to me too

**Pull request merged**